### DISCUSSION TITLE
Which docker image to use?

### DISCUSSION THREAD

**Discussion by Dunedan** [processone/ejabberd#3880] [Questions] [answered]
In the release notes for ejabberd 22.05 a new Docker image is mentioned (https://github.com/processone/ejabberd/releases/tag/22.05), which is documented in https://github.com/processone/ejabberd/blob/master/CONTAINER.md.

While the release notes point out the differences to the existing `ejabberd/ecs` image, I miss some guidance which image to use going forward. Is one of these images the preferred one to use? Will `ejabberd/ecs` continue to be maintained? Are there any other differences to expect between the images in future?

**Comment by badlop** [comment, marked as answer]
> While the release notes point out the differences to the existing ejabberd/ecs image, I miss some guidance which image to use going forward.

The GHCR image was introduced in the last release to accomplish three things: arm64 support, automated build and publish, support for nightly `master` images, and require less customizations. Probably all this would have been possible with Docker Hub... but would have required more effort, and it was preferable to not modify `ejabberd/ecs`.


> Is one of these images the preferred one to use?

The Docker image is available since many years ago, so it won't disappear or receive major changes: it must keep stable and trusted.

The GHCR image was added recently, it's easier to improve, test, so it's more probable to receive improvements.


> Will ejabberd/ecs continue to be maintained?

Yes. It's being used (at least statistics say so), requires only some manual effort to publish it... no reason to deprecate it.

And if it were deprecated, the change would be just to be a verbatim copy of the GHCR (if someday that one is really preferable), with a migration tutorial and several months/releases announcement in advance so administrators can plan it.

> Are there any other differences to expect between the images in future?

Well, I don't know other people, but I find easier to develop and test improvements in GHCR than Docker Hub: I just fork ejabberd repo, create a branch, implement features in container/, and automatically a new image is built for me or others to test.

Whatever new features, I'll try them first in GHCR, and maybe later adapt to the Docker image. The first one is an option to support epmd-less (no need to open ports 4369 and a range of other ports to support clustering).

Basically, I expect new features to be available for sure in the GHCR image, and appearance in Docker Hub is not guaranteed.

**Reply by Dunedan** [reply]
Thanks a lot for your answer. I'll proceed with the GHCR image then.

**Comment by YazeedAlKhalaf** [comment]
I used the GHCR image but the `conf` folder location is different that the one on docker. Is that normal?

GHCR image has it under `/opt/ejabberd/conf`
Docker Hub image has is under /home/ejabberd/conf

**Reply by badlop** [reply]
Right, this and other differences are enumerated in [Alternative Image in GitHub](https://github.com/processone/docker-ejabberd/blob/master/ecs/HUB-README.md#alternative-image-in-github)
